bin.src/exportGraphInputs.py

```diff
@@ -48,27 +48,25 @@
     os.makedirs(args.output, exist_ok=True)
     with butler.export(directory=args.output, format="yaml", transfer="auto") as export:
         items = []
-        for graph_node in graph:
-            for quantum in graph_node.quanta:
-                for key, value in quantum.predictedInputs.items():
-                    for dataset in value:
-                        if dataset.datasetType.name not in dataset_types_to_exclude:
-                            # The quantum graph doesn't know the ID of the
-                            # real dataset so convert to a real ref
-                            found = set(butler.registry.queryDatasets(dataset.datasetType.name,
-                                                                      collections=...,
-                                                                      dataId=dataset.dataId))
-                            items.extend(found)
+        for quantum_node in graph:
+            for key, value in quantum_node.quantum.inputs.items():
+                for dataset in value:
+                    if dataset.datasetType.name not in dataset_types_to_exclude:
```

Comment:
`quantum.inputs` is already a mapping of `DatasetType`. As it is now, `key` is not used. If this was rearranged to:
```
...
for key, value in quantum_node.quantum.inputs.items():
    if key.name not in dataset_types_to_exclude:
        found = {butler.registiry.queryDatasets(key.name, collections=..., dataId=dataset.dataId) for dataId in value}
...
```
It would save a for loop over a lot of elements that will be excluded by `dataset_types_to_exclude`

Reply:
I have rejigged a bit based on the suggestion above but I can't use that line that does the loop over refs in queryDatasets because queryDatasets returns an interator and not DatasetRef and so you end up with a set of iterators. It's easier to use still use an explicit loop over refs (but still inside the if-check).

One thing that does confuse me is that the `quantum_node.quantum.inputs` is returning a proper DatasetRef with the `id` attached so the reality is that this queryDataset call is returning the same thing it had to start with. Not entirely sure that we expect that or not so maybe it's a special case for this particular example.